libcxx/include/__exception/exception_ptr.h

```diff
@@ -51,11 +69,34 @@ class _LIBCPP_EXPORTED_FROM_ABI exception_ptr {
 template <class _Ep>
 _LIBCPP_HIDE_FROM_ABI exception_ptr make_exception_ptr(_Ep __e) _NOEXCEPT {
 #  ifndef _LIBCPP_HAS_NO_EXCEPTIONS
+#    if defined(LIBCXXRT) || defined(LIBCXX_BUILDING_LIBCXXABI)
+  using _Ep2 = __decay_t<_Ep>;
+  void* __ex = exception_ptr::__init_native_exception(
+      sizeof(_Ep), const_cast<std::type_info*>(&typeid(_Ep)), exception_ptr::__dest_thunk<_Ep2>);
```

Comment:
Can't use `typeid` with -fno-rtti, but exceptions can be used with -fno-rtti so the previous impl would work. So this code should all be in a !_LIBCPP_HAS_NO_RTTI ifdef.

Reply:
> Can't use `typeid` with -fno-rtti, but exceptions can be used with -fno-rtti so the previous impl would work. So this code should all be in a !_LIBCPP_HAS_NO_RTTI ifdef.

Hmm... (at least on Itanium ABI?) when exceptions need to be thrown, `type_info` objects are generated and passed on throwing even if -fno-rtti is used.

Is there any intrinsic to obtain RTTI for EH when -fno-rtti is used?